src/java.desktop/windows/classes/sun/print/Win32PrintJob.java

```diff
@@ -437,18 +433,7 @@ public void print(Doc doc, PrintRequestAttributeSet attributes)
 
             if (mDestination != null) { // if destination attribute is set
                 try {
-                    FileOutputStream fos = new FileOutputStream(mDestination);
-                    byte []buffer = new byte[1024];
-                    int cread;
-
-                    while ((cread = instream.read(buffer, 0, buffer.length)) >=0) {
-                        fos.write(buffer, 0, cread);
-                    }
-                    fos.flush();
-                    fos.close();
-                } catch (FileNotFoundException fnfe) {
-                    notifyEvent(PrintJobEvent.JOB_FAILED);
-                    throw new PrintException(fnfe.toString());
+                    Files.copy(instream, Path.of(mDestination), StandardCopyOption.REPLACE_EXISTING);
```

Comment:
Don't we need to close the instream in finally block?

Reply:
It will be closed in `notifyEvent(PrintJobEvent.JOB_FAILED);` -> `closeDataStreams` in case of IOException.
And there shouldn't be any other kinds of exceptions.